assets/src/components/template-mode-option/index.js

```diff
@@ -75,59 +85,78 @@ function getTitle( mode ) {
  * @param {string} props.details Mode details.
  * @param {string} props.mode The template mode.
  * @param {boolean} props.previouslySelected Optional. Whether the option was selected previously.
+ * @param {Object} props.labelExtra Optional. Extra content to display on the right side of the option label.
+ * @param {boolean} props.initialOpen Whether the panel should be open when the component renders.
  */
-export function TemplateModeOption( { children, details, mode, previouslySelected = false } ) {
+export function TemplateModeOption( { children, details, initialOpen, labelExtra = null, mode, previouslySelected = false } ) {
 	const { editedOptions, updateOptions } = useContext( Options );
 	const { theme_support: themeSupport } = editedOptions;
 
-	const id = `template-mode-${ mode }`;
+	const id = getId( mode );
 
 	return (
-		<Selectable id={ `${ id }-container` } className="template-mode-option" selected={ mode === themeSupport }>
-			<label htmlFor={ id }>
-				<div className="template-mode-selection__input-container">
-					<input
-						type="radio"
-						id={ id }
-						checked={ mode === themeSupport }
-						onChange={ () => {
-							updateOptions( { theme_support: mode } );
-						} }
-					/>
-				</div>
-				<div className="template-mode-selection__illustration">
-					{ <Illustration mode={ mode } /> }
-				</div>
-				<div className="template-mode-selection__description">
-					<h2>
-						{ getTitle( mode ) }
-					</h2>
-					{ previouslySelected && (
-						<AMPInfo>
-							{ __( 'Previously selected', 'amp' ) }
-						</AMPInfo>
-					) }
-				</div>
-			</label>
+		<AMPDrawer
+			className="template-mode-option"
+			handleType={ HANDLE_TYPE_RIGHT }
+			heading={ (
+				<label className="template-mode-option__label" htmlFor={ id }>
+					<div className="template-mode-selection__input-container">
+						<input
+							type="radio"
+							id={ id }
+							checked={ mode === themeSupport }
+							onChange={ () => {
+								updateOptions( { theme_support: mode } );
+							} }
+						/>
+					</div>
+					<div className="template-mode-selection__illustration">
+						{ <Illustration mode={ mode } /> }
+					</div>
+					<div className="template-mode-selection__description">
+						<h3>
+							{ getTitle( mode ) }
+						</h3>
+						{ previouslySelected && (
+							<AMPInfo>
+								{ __( 'Previously selected', 'amp' ) }
+							</AMPInfo>
+						) }
+						{ labelExtra && (
+							<div className="template-mode-selection__label-extra">
+								{ labelExtra }
+							</div>
+						) }
+					</div>
+				</label>
+			) }
+			hiddenTitle={ getTitle( mode ) }
+			id={ `${ id }-container` }
+			initialOpen={ 'boolean' === typeof initialOpen ? initialOpen : mode && themeSupport && mode === themeSupport }
```

Comment:
`initialOpen` would always be a boolean, no?

```suggestion
			initialOpen={ initialOpen || ( mode && themeSupport ) && mode === themeSupport }
```